common/client.go

```diff
@@ -22,10 +47,27 @@ type DatabricksClient struct {
 
 	// callback used to create API1.2 call wrapper, which simplifies unit tessting
 	commandFactory func(context.Context, *DatabricksClient) CommandExecutor
+	cachedUser     scim.User
```

Comment:
```suggestion
	cachedUser     *scim.User
```

Reply:
This would cause caching to not propagate. The explanation https://github.com/databricks/terraform-provider-databricks/pull/2170#discussion_r1151711693